platform/platformutils.go

```diff
@@ -0,0 +1,248 @@
+			},
+		},
+	}
+
+	err := client.Create(ctx, cm)
+	if err != nil && !k8serrors.IsAlreadyExists(err) {
+		return nil, err
+	}
+
+	return cm, nil
+}
+
+func SetPlatformDefaults(p *v08.KogitoServerlessPlatform, verbose bool) error {
+	if p.Status.BuildPlatform.PublishStrategyOptions == nil {
+		log.Debugf("Kogito Serverless Platform [%s]: setting publish strategy options", p.Namespace)
+		p.Status.BuildPlatform.PublishStrategyOptions = map[string]string{}
+	}
+
+	if _, ok := p.Status.BuildPlatform.PublishStrategyOptions[builder.KanikoPVCName]; !ok {
+		log.Debugf("Kogito Serverless Platform [%s]: setting publish strategy options", p.Namespace)
+		p.Status.BuildPlatform.PublishStrategyOptions[builder.KanikoPVCName] = p.Name
+	}
+
+	if p.Status.BuildPlatform.GetTimeout().Duration != 0 {
+		d := p.Status.BuildPlatform.GetTimeout().Duration.Truncate(time.Second)
+
+		if verbose && p.Status.BuildPlatform.GetTimeout().Duration != d {
+			log.Log.Infof("Build timeout minimum unit is sec (configured: %s, truncated: %s)", p.Status.BuildPlatform.GetTimeout().Duration, d)
+		}
+
+		log.Debugf("Kogito Serverless Platform [%s]: setting build timeout", p.Namespace)
+		p.Status.BuildPlatform.Timeout = &metav1.Duration{
+			Duration: d,
+		}
+	}
+	if p.Status.BuildPlatform.GetTimeout().Duration == 0 {
+		p.Status.BuildPlatform.Timeout = &metav1.Duration{
+			Duration: 5 * time.Minute,
+		}
+	}
+	_, cacheEnabled := p.Status.BuildPlatform.PublishStrategyOptions[builder.KanikoBuildCacheEnabled]
+	if p.Status.BuildPlatform.PublishStrategy == api.PlatformBuildPublishStrategyKaniko && !cacheEnabled {
+		// Default to disabling Kaniko cache warmer
+		// Using the cache warmer pod seems unreliable with the current Kaniko version
+		// and requires relying on a persistent volume.
+		defaultKanikoBuildCache := "false"
+		p.Status.BuildPlatform.PublishStrategyOptions[builder.KanikoBuildCacheEnabled] = defaultKanikoBuildCache
+		if verbose {
+			log.Log.Infof("Kaniko cache set to %s", defaultKanikoBuildCache)
+		}
+	}
+
+	setStatusAdditionalInfo(p)
+
+	if verbose {
+		log.Log.Infof("BaseImage set to %s", p.Status.BuildPlatform.BaseImage)
+		log.Log.Infof("Timeout set to %s", p.Status.BuildPlatform.GetTimeout())
+	}
+	return nil
+}
+
+func setStatusAdditionalInfo(platform *v08.KogitoServerlessPlatform) {
+	platform.Status.Info = make(map[string]string)
+
+	log.Debugf("Kogito Serverless Platform [%s]: setting build publish strategy", platform.Namespace)
+	if platform.Spec.BuildPlatform.PublishStrategy == api.PlatformBuildPublishStrategyKaniko {
+		platform.Status.Info["kanikoVersion"] = defaults.KanikoVersion
+	}
+	log.Debugf("Kogito Serverless [%s]: setting status info", platform.Namespace)
+	platform.Status.Info["goVersion"] = runtime.Version()
+	platform.Status.Info["goOS"] = runtime.GOOS
+}
+
+// IsOpenShift returns true if we are connected to a OpenShift cluster.
+func IsOpenShift(client k8s.Interface) (bool, error) {
```

Comment:
there is a useful function in the oeprator-utils to retrieve this info.

Reply:
As discussed we can introduce the usage of the operator-utils after releasing a version that we can use into the operator.